internal/backend/local/backend_local.go

```diff
@@ -187,11 +187,13 @@ func (b *Local) localRunDirect(op *backend.Operation, run *backend.LocalRun, cor
 	}
 
 	planOpts := &terraform.PlanOpts{
-		Mode:         op.PlanMode,
-		Targets:      op.Targets,
-		ForceReplace: op.ForceReplace,
-		SetVariables: variables,
-		SkipRefresh:  op.Type != backend.OperationTypeRefresh && !op.PlanRefresh,
+		Mode:            op.PlanMode,
+		Targets:         op.Targets,
+		ForceReplace:    op.ForceReplace,
+		SetVariables:    variables,
+		SkipRefresh:     op.Type != backend.OperationTypeRefresh && !op.PlanRefresh,
+		AutoApprove:     op.AutoApprove,
+		PlanDuringApply: op.Type == backend.OperationTypeApply,
```

Comment:
When is a `PlanDuringApply` not an `AutoApprove` operation? It looks like these are both only checked in the same conditional expression.

It would be nicer if we could lift this logic up a little higher still, out of the planning process entirely, so that the plan never acts any differently regardless of `-auto-approve`, and it's only handled by the presentation logic. Perhaps the diagnostics could be filtered at the end by `views`?

Reply:
So any plan that occurs during `terraform apply` without a plan file would meet the criteria for PlanDuringApply, vs AutoApprove which explicitly needs the `-auto-approve` flag set.

Having said that it doesn't matter anymore. I didn't move this into the view because we'd need a lot of extra information into the view package that's not currently needed, but I did manage to find exactly where the apply operation decides whether it's going to ask for approval, and then added my filtering logic there. So the filtering logic no longer cares about why it's not asking for approval, it simply knows whether or not it has which I think is better and has moved the logic another level up which should hopefully enough.

I could go ahead and add it into the views package but then we'd need to pass the plan or a confirmation flag somewhere in there which seems unnecessary when we have the logic packaged up nicely already.